src/components/accession2/CalculatorModal.tsx

```diff
@@ -0,0 +1,138 @@
+import React from 'react';
+import strings from 'src/strings';
+import Button from '../common/button/Button';
+import DialogBox from '../common/DialogBox/DialogBox';
+import { Grid } from '@mui/material';
+import { SelectT, Textfield } from '@terraware/web-components';
+import { Accession2, updateAccession2 } from 'src/api/accessions2/accession';
+import { ServerOrganization } from 'src/types/Organization';
+import { Unit, WEIGHT_UNITS_V2 } from '../seeds/nursery/NewTest';
+import useSnackbar from 'src/utils/useSnackbar';
+
+export interface CalculatorDialogProps {
+  open: boolean;
+  record: Accession2;
+  setRecord: React.Dispatch<React.SetStateAction<Accession2>>;
+  onClose: () => void;
+  reload: () => void;
+  organization: ServerOrganization;
+  onChange: (id: string, value: unknown) => void;
+  openPreviousModal: () => void;
+}
+
+export default function CalculatorDialog(props: CalculatorDialogProps): JSX.Element {
+  const { onClose, open, record, onChange, setRecord, openPreviousModal } = props;
+
+  const snackbar = useSnackbar();
+
+  const getTotalCount = async () => {
+    const response = await updateAccession2(record);
```

Comment:
this updates the accession in the database,, is that what we want? i thought this modal just does some calculation for us.

Reply:
Yes, we need to get the estimated seed count value, and for doing that we have to save values and get the value from the api, but now that you mention I forgot to reset it back to original values if you press cancel. Will do that

Reply:
Ok, i think we can ask Steve for a preview api .. if user does not click cancel and closes browser we have modified data.

Reply:
I was going to save the original values and restore them (save the accession again with those values) before onClose (cancel or clicking outside call the onClose method). But it sounds cleaner to have an api for getting that value.

Reply:
@constanzauanini actually, you can use the 'simulate: true' option .. the API will return the values but won't update data on disk. We just need to pass `?simulate=true` param with the API

Reply:
Oh, that's true! I forgot about that! Excellent! Thank you